plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/converter/GlueStatConverter.java

```diff
@@ -0,0 +1,291 @@
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.getTotalSizeInBytes;
+import static io.trino.plugin.hive.metastore.thrift.ThriftMetastoreUtil.toMetastoreDistinctValuesCount;
+import static org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category.PRIMITIVE;
+
+public class GlueStatConverter
+{
+    private GlueStatConverter() {}
+
+    private static final long MILLIS_PER_DAY = TimeUnit.DAYS.toMillis(1);
+
+    public static List<ColumnStatistics> toGlueColumnStatistics(
+            Partition partition, Map<String, HiveColumnStatistics> trinoColumnStats, OptionalLong rowCount)
+    {
+        return partition.getColumns().stream()
+                .filter(column -> trinoColumnStats.containsKey(column.getName()))
+                .map(c -> toColumnStatistics(c, trinoColumnStats.get(c.getName()), rowCount))
+                .collect(toImmutableList());
+    }
+
+    public static List<ColumnStatistics> toGlueColumnStatistics(
+            Table table, Map<String, HiveColumnStatistics> trinoColumnStats, OptionalLong rowCount)
+    {
+        return trinoColumnStats.entrySet().stream()
+                .map(e -> toColumnStatistics(table.getColumn(e.getKey()).get(), e.getValue(), rowCount))
+                .collect(toImmutableList());
+    }
+
+    private static ColumnStatistics toColumnStatistics(Column column, HiveColumnStatistics statistics, OptionalLong rowCount)
+    {
+        ColumnStatistics columnStatistics = new ColumnStatistics();
+        HiveType columnType = column.getType();
+        columnStatistics.setColumnName(column.getName());
+        columnStatistics.setColumnType(columnType.toString());
+        ColumnStatisticsData catalogColumnStatisticsData = toGlueColumnStatisticsData(statistics, columnType, rowCount);
+        columnStatistics.setStatisticsData(catalogColumnStatisticsData);
+        columnStatistics.setAnalyzedTime(new Date());
+        return columnStatistics;
+    }
+
+    public static HiveColumnStatistics fromGlueColumnStatistics(ColumnStatisticsData catalogColumnStatisticsData, OptionalLong rowCount)
+    {
+        ColumnStatisticsType type = ColumnStatisticsType.fromValue(catalogColumnStatisticsData.getType());
+        switch (type) {
+            case BINARY:
+                BinaryColumnStatisticsData catalogBinaryData = catalogColumnStatisticsData.getBinaryColumnStatisticsData();
+                OptionalLong maxColumnLengthOfBinary = OptionalLong.of(catalogBinaryData.getMaximumLength());
+                OptionalDouble averageColumnLengthOfBinary = OptionalDouble.of(catalogBinaryData.getAverageLength());
+                OptionalLong nullsCountOfBinary = fromMetastoreNullsCount(catalogBinaryData.getNumberOfNulls());
+                return createBinaryColumnStatistics(
+                        maxColumnLengthOfBinary,
+                        getTotalSizeInBytes(averageColumnLengthOfBinary, rowCount, nullsCountOfBinary),
+                        nullsCountOfBinary);
```

Comment:
the `nullsCountOfBinary` variable is the result of `fromMetastoreNullsCount(catalogBinaryData.getNumberOfNulls());` in this method is assigned to a variable because the value is used twice